src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -309,8 +309,20 @@ static int _PaPulseAudio_ProcessAudio(PaPulseAudio_Stream *stream,
 
     if( !stream->isActive && stream->pulseaudioIsActive && stream->outputStream)
     {
-        bufferData = pulseaudioSampleBuffer;
-        memset( bufferData, 0x00, length);
+        size_t tmpSize = length;
+
+        /* Allocate memory to make it faster to output stuff */
+        pa_stream_begin_write( stream->outputStream, &bufferData, &tmpSize );
+
+        /* If bufferData is NULL then output is not ready
+         * and we have to wait for it
+         */
+        if(!bufferData)
+        {
+            return paNotInitialized;
```

Comment:
Using `paNotInitialized` here is a bit unusual but I guess it's OK if you're checking for this value in the caller.

Reply:
What would be better. It's just error for caller but I also thought it's no good but could not figure out what should be used. Should it be just -1 or something for error.. too fancy.

Reply:
Yes, I think -1 would be better. Or define your own internal error codes that capture the actual meaning of the error conditions (or use pulse audio error codes if that is feasible. I notice that you never check the return codes.

Reply:
Ok this should be now fixed